src/clint.py

```diff
@@ -2600,16 +2600,23 @@ def CheckBraces(filename, clean_lines, linenum, error):
         else:
             func_start_linenum = end_linenum + 1
             while not clean_lines.lines[func_start_linenum] == '{':
-                if not Match(r'^(?:\s*\b(?:FUNC_ATTR|REAL_FATTR)_\w+\b(?:\(\d+(, \d+)*\))?)+$',
-                             clean_lines.lines[func_start_linenum]):
+                attrline = Match(r'^(\s*)(?:FUNC_ATTR|FUNC_API|REAL_FATTR)_\w+(?:\(\d+(, \d+)*\))?',
```

Comment:
@justinmk What about making first group `((?!# *define).*)` and use `attrline.group(1) != '  '` below? This will ban FUNC_ATTR/etc placed like this:

```C
int foo(const int bar) FUNC_ATTR_PURE
```

, requiring attributes always be on their own line. I find that more readable.